### DISCUSSION TITLE
New UI & Function Suggestions

### DISCUSSION THREAD

**Discussion by AkaXiiaoBii** [AkaXiiaoBii/mdoujin_v2_public_updates#7] [Ideas]
Hello, Akabii here,

I am planning to develop a new UI in the future time (Estimate July 2023) will start to develop it.
Anyone who wants to purpose new features can leave a comment here. 

Thank you.
Akabii

**Comment by AkaXiiaoBii** [comment]
New features:

1. Remain most of the current base features, and redesign the new UI.

2. Event List 
-Allow adding banners/posters for every event.
-Event organizer selection.
-Download multiple calendar events.
-Bulk download calendar(.ics) based on the displayed event list.
-Add event - Event information field support HTML editor format (low priority) something as below:
![image](https://github.com/AkaXiiaoBii/mdoujin_v2_public_updates/assets/29534833/ac4cd2b1-c5ca-4d6a-9e55-3729892460ef)
-Add Ticketing Button for visitors easy access to the ticketing site for purchasing.
-Add (Cosplay Plan Template Hyperlink ) to let the user download the cosplay plan template. (Added on 14 June 2023)
-Support thumbnail view and list view with switching button. (Added on 4 July 2023)

3. Event Organizer List
-Event Organizer history, estimated year, logos, etc
-Each Event Organizer will display their own relevant upcoming/past events

4. Official tickets/booklet picture upload collections
- Visitors can share past event tickets/booklets. Upon Administrator approval, will be published on the website for display

5. Allow Enable/Disable Dark Mode (Low Priority)

Latest updated: 14 June 2023

**Comment by KuroShion** [comment]
Suggested Feature:

1. Add Ticketing Button for visitor easy access to ticketing site for purchasing
2. Dark Mode
3. Maybe ,(Just Maybe) Add a voting function for visitor rate the event

**Reply by AkaXiiaoBii** [reply]
> Suggested Feature:
> 
> 1. Add Ticketing Button for visitor easy access to ticketing site for purchasing
> 2. Dark Mode
> 3. Maybe ,(Just Maybe) Add a voting function for visitor rate the event

Hi EvoTeamMalaysia, 

Noted, 
1 & 2 agreed and considered for this development plan.
For 3, not at the moment since this might be getting abused it doesn't have proper control, btw it's a good suggestion.

**Comment by Kiiingsin** [comment]
Will Mdoujin release App version？

**Reply by AkaXiiaoBii** [reply]
At the moment will not have plan on developing mobile app, so this will be web application (as currently mdoujin using right now).

**Comment by jerrywoo96** [comment]
allow switching between thumbnail view (if that is you want to implement thumbnail view like shopee) and list view, as list view is more suitable  and easier on eyes and brain processing for chronological viewing. For example, you can use the url provided as inspiration, as i find that their design is very intuitive.

https://www.bhphotovideo.com/c/buy/Mirrorless-Camera-Lenses/ci/17912/N/4196380428

**Reply by jerrywoo96** [reply]
There is a change view on the top right hand corner of the table of listings.
![image](https://github.com/AkaXiiaoBii/mdoujin_v2_public_updates/assets/26324139/64d80e4f-afee-41a6-9a4f-5f6eed7444f8)

**Reply by jerrywoo96** [reply]
apologies in advance, as i only just saw this discussion now.

**Reply by jerrywoo96** [reply]
Thumbnail grid view like shopee is more for an overall type of view to see what events are available at a quick glance and how many events are there.

**Reply by AkaXiiaoBii** [reply]
Hi, Jerry Woo, thanks for your message and suggestion,

For event listing, I am planning to develop both thumbnail view (layout will be something like https://osu.ppy.sh/beatmapsets) and simple list (as currently mdoujin event list with more enhanced version, sometimes users only want to see the summarised event list for easy to screenshot.). 

Will be recorded for this enhancement plan too, thank you.

**Reply by jerrywoo96** [reply]
oh, and the reason why last time i requested an api was because i wanted to make my own ical endpoint. If ical is developed, i do not need the api in fact. currently, i have a working ical file based off the data from the website, do keep in touch with me on fb if you would like a sample of the ical.

**Reply by jerrywoo96** [reply]
the ical endpoint should be able to use same querystring as the website.

**Reply by jerrywoo96** [reply]
an alternative is, i can host the ical endpoint publicly for you, if you can provide me a private access to json api using basic auth or jwt or api key, and we can do a cname redirect to get the ical file for the community. my server is protected behind cloudflare. the json api can be as simple as using the same querystring as the web version.

**Reply by jerrywoo96** [reply]
so my server will bear all the requests for the ical endpoint.

**Reply by jerrywoo96** [reply]
i cannot assure you of the server uptime, as its a homelab server, but hosting a useful endpoint still beats having no endpoint at all. my goal is to be able to to use this ical endpoint, if i'm unable to host the endpoint in the future, we can discuss again how to migrate.

**Reply by jerrywoo96** [reply]
i'm using flask in python docker container to run the endpoint.
